torchtune/data/tokenizers/_tiktoken.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, Iterator, List
+
+from tiktoken import Encoding
+from tiktoken.load import load_tiktoken_bpe
+from torchtune.data.tokenizers._base import TokenEncoding
+
+# Constants controlling encode logic
+MAX_ENCODE_CHARS = 400_000
+MAX_NO_WHITESPACE_CHARS = 25_000
+
+
+class TikTokenEncoding(TokenEncoding):
+    """
+    A lightweight wrapper around tiktoken Encoding. This class additionally handles
+    breaking up the input text into substrings of a max length and splitting up long
+    repetitions to improve encode speed.
+
+    The core tiktoken.Encoding does NOT natively handle BOS and EOS ids, so this class
+    does not either. They must be handled by the caller of ``encode``.
```

Comment:
This is a bit confusing to me (I understand the motivation though). In practice I would strive for more consistency in the `encode` APIs -- obviously there are some pretty custom things in sentencepiece that we don't wanna recreate  across the board, so not saying the signatures should match exactly or anything. But BOS and EOS handling should imo be exposed in the same place

Reply:
I'm actually not sure why I didn't just move EOS and BOS handling to the TikToken class..